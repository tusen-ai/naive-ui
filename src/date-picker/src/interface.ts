import type { Ref, Slots, UnwrapNestedRefs } from 'vue'
import { type VirtualListInst } from 'vueuc'
import type { NLocale, NDateLocale } from '../../locales'
import type { ScrollbarInst } from '../../_internal'
import type {
  IsHourDisabled,
  IsMinuteDisabled,
  IsSecondDisabled
} from '../../time-picker/src/interface'
import type { TimePickerProps } from '../../time-picker/src/TimePicker'
import type { MergedTheme } from '../../_mixins'
import { createInjectionKey } from '../../_utils'
import type { DatePickerTheme } from '../styles/light'
import type {
  uniCalendarValidation,
  dualCalendarValidation
} from './validation-utils'

export type Value = number | [number, number]

export type DefaultTime = string | [string | undefined, string | undefined]

export type FormattedValue = string | [string, string]

export type Shortcuts =
  | Record<string, number | (() => number)>
  | Record<
  string,
  | [number, number]
  | readonly [number, number]
  | (() => [number, number] | readonly [number, number])
  >

export type OnUpdateValue = (
  value: number &
  (number | null) &
  [number, number] &
  ([number, number] | null),
  formattedValue: string &
  (string | null) &
  [string, string] &
  ([string, string] | null)
) => void

export type OnConfirm = OnUpdateValue

export type OnConfirmImpl = OnUpdateValueImpl

export type OnUpdateFormattedValue = (
  value: string &
  (string | null) &
  [string, string] &
  ([string, string] | null),
  timestampValue: number &
  (number | null) &
  [number, number] &
  ([number, number] | null)
) => void

export type OnUpdateFormattedValueImpl = (
  value: string | [string, string] | null,
  timestampValue: number | [number, number] | null
) => void

export type OnUpdateValueImpl = (
  value: Value | null,
  formattedValue: string | [string, string] | null
) => void

export type OnPanelUpdateValue = (
  value: number &
  (number | null) &
  [number, number] &
  ([number, number] | null),
  doUpdate: boolean
) => void

export type OnPanelUpdateValueImpl = (
  value: Value | null,
  doUpdate: boolean
) => void

export type OnClose = (disableUpdateOnClose: boolean) => void

export interface RangePanelChildComponentRefs {
  startYearScrollbarRef: Ref<ScrollbarInst | null>
  endYearScrollbarRef: Ref<ScrollbarInst | null>
  startMonthScrollbarRef: Ref<ScrollbarInst | null>
  endMonthScrollbarRef: Ref<ScrollbarInst | null>
  startYearVlRef: Ref<VirtualListInst | null>
  endYearVlRef: Ref<VirtualListInst | null>
}

export interface PanelChildComponentRefs {
  monthScrollbarRef: Ref<ScrollbarInst | null> // Only exists when type is month
  yearScrollbarRef: Ref<ScrollbarInst | null>
  // year, virtual scroll
  yearVlRef: Ref<VirtualListInst | null>
}

export interface PanelRef
  extends Partial<
  UnwrapNestedRefs<PanelChildComponentRefs & RangePanelChildComponentRefs>
  > {
  $el: HTMLElement
}

// 0 is Monday
export type FirstDayOfWeek = 0 | 1 | 2 | 3 | 4 | 5 | 6

export type DatePickerInjection = {
  mergedClsPrefixRef: Ref<string>
  mergedThemeRef: Ref<MergedTheme<DatePickerTheme>>
  timePickerSizeRef: Ref<'small' | 'medium' | 'large'>
  timePickerPropsRef: Ref<
  undefined | TimePickerProps | [TimePickerProps, TimePickerProps]
  >
  localeRef: Ref<NLocale['DatePicker']>
  dateLocaleRef: Ref<NDateLocale>
  isDateDisabledRef: Ref<IsDateDisabled | undefined>
  rangesRef: Ref<Record<string, [number, number]> | undefined>
  closeOnSelectRef: Ref<boolean>
  updateValueOnCloseRef: Ref<boolean>
  firstDayOfWeekRef: Ref<FirstDayOfWeek | undefined>
  datePickerSlots: Slots
} & ReturnType<typeof uniCalendarValidation> &
ReturnType<typeof dualCalendarValidation>

export const datePickerInjectionKey =
  createInjectionKey<DatePickerInjection>('n-date-picker')

export type IsDateDisabled = IsSingleDateDisabled | IsRangeDateDisabled
export type IsSingleDateDisabled = (date: number) => boolean
export type IsRangeDateDisabled = (
  date: number,
  position: 'start' | 'end',
  value: [number, number] | null
) => boolean

export interface TimeValidator {
  isHourDisabled?: IsHourDisabled
  isMinuteDisabled?: IsMinuteDisabled
  isSecondDisabled?: IsSecondDisabled
}

export type IsTimeDisabled = IsSingleTimeDisabled | IsRangeTimeDisabled
export type IsSingleTimeDisabled = (date: number) => TimeValidator
export type IsRangeTimeDisabled = (
  date: number,
  position: 'start' | 'end',
  value: [number, number] // date must exist to have time validation
) => TimeValidator

export interface DatePickerInst {
  focus: () => void
  blur: () => void
}
