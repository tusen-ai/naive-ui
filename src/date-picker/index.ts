export { default as NDatePicker } from './src/DatePicker'
export { datePickerProps } from './src/props'
export type * from './src/public-types'
