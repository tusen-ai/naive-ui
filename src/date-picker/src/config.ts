export const START_YEAR = 1901
// TODO: we need to remove it to make height customizable
export const MONTH_ITEM_HEIGHT = 40

export type DatePickerType =
  | 'date'
  | 'datetime'
  | 'daterange'
  | 'datetimerange'
  | 'month'
  | 'year'
  | 'quarter'
  | 'monthrange'
  | 'quarterrange'
  | 'yearrange'
  | 'week'
