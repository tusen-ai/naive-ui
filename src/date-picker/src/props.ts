import type { PropType } from 'vue'
import type { FollowerPlacement } from 'vueuc/es'
import type { ThemeProps } from '../../_mixins'
import { useTheme } from '../../_mixins'
import type { MaybeArray } from '../../_utils'
import { useAdjustedTo } from '../../_utils'
import type { DatePickerTheme } from '../styles'
import type { FormValidationStatus } from '../../form/src/interface'
import type { TimePickerProps } from '../../time-picker'
import type {
  DefaultTime,
  FirstDayOfWeek,
  FormattedValue,
  IsDateDisabled,
  IsTimeDisabled,
  OnConfirm,
  OnUpdateFormattedValue,
  OnUpdateValue,
  Shortcuts,
  Value
} from './interface'
import type { DatePickerType } from './config'

export const datePickerProps = {
  ...(useTheme.props as ThemeProps<DatePickerTheme>),
  to: useAdjustedTo.propTo,
  bordered: {
    type: Boolean as PropType<boolean | undefined>,
    default: undefined
  },
  clearable: Boolean,
  updateValueOnClose: Boolean,
  defaultValue: [Number, Array] as PropType<Value | null>,
  defaultFormattedValue: [String, Array] as PropType<FormattedValue | null>,
  defaultTime: [Number, String, Array] as PropType<DefaultTime>,
  disabled: {
    type: Boolean as PropType<boolean | undefined>,
    default: undefined
  },
  placement: {
    type: String as PropType<FollowerPlacement>,
    default: 'bottom-start'
  },
  value: [Number, Array] as PropType<Value | null>,
  formattedValue: [String, Array] as PropType<FormattedValue | null>,
  size: String as PropType<'small' | 'medium' | 'large'>,
  type: {
    type: String as PropType<DatePickerType>,
    default: 'date'
  },
  valueFormat: String,
  separator: String,
  placeholder: String,
  startPlaceholder: String,
  endPlaceholder: String,
  format: String,
  dateFormat: String,
  timerPickerFormat: String,
  actions: Array as PropType<Array<'clear' | 'confirm' | 'now'> | null>,
  shortcuts: Object as PropType<Shortcuts>,
  isDateDisabled: Function as PropType<IsDateDisabled>,
  isTimeDisabled: Function as PropType<IsTimeDisabled>,
  show: {
    type: Boolean as PropType<boolean | undefined>,
    default: undefined
  },
  panel: Boolean,
  ranges: Object as PropType<Record<string, [number, number]>>,
  firstDayOfWeek: Number as PropType<FirstDayOfWeek>,
  inputReadonly: Boolean,
  closeOnSelect: Boolean,
  status: String as PropType<FormValidationStatus>,
  timePickerProps: [Object, Array] as PropType<
    TimePickerProps | [TimePickerProps, TimePickerProps]
  >,
  onClear: Function as PropType<() => void>,
  onConfirm: Function as PropType<OnConfirm>,
  defaultCalendarStartTime: Number,
  defaultCalendarEndTime: Number,
  bindCalendarMonths: Boolean,
  monthFormat: { type: String, default: 'M' },
  yearFormat: { type: String, default: 'y' },
  quarterFormat: { type: String, default: '\'Q\'Q' },
  yearRange: {
    type: Array as unknown as PropType<[number, number]>,
    default: (): [number, number] => [1901, 2100]
  },
  'onUpdate:show': [Function, Array] as PropType<
    MaybeArray<(show: boolean) => void>
  >,
  onUpdateShow: [Function, Array] as PropType<
    MaybeArray<(show: boolean) => void>
  >,
  'onUpdate:formattedValue': [Function, Array] as PropType<
    MaybeArray<OnUpdateFormattedValue>
  >,
  onUpdateFormattedValue: [Function, Array] as PropType<
    MaybeArray<OnUpdateFormattedValue>
  >,
  'onUpdate:value': [Function, Array] as PropType<MaybeArray<OnUpdateValue>>,
  onUpdateValue: [Function, Array] as PropType<MaybeArray<OnUpdateValue>>,
  onFocus: [Function, Array] as PropType<(e: FocusEvent) => void>,
  onBlur: [Function, Array] as PropType<(e: FocusEvent) => void>,
  onNextMonth: Function as PropType<() => void>,
  onPrevMonth: Function as PropType<() => void>,
  onNextYear: Function as PropType<() => void>,
  onPrevYear: Function as PropType<() => void>,
  // deprecated
  onChange: [Function, Array] as PropType<MaybeArray<OnUpdateValue>>
} as const
