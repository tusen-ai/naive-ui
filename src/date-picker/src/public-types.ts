import type { ExtractPublicPropTypes } from '../../_utils'
import type { datePickerProps } from './props'

export interface DatePickerInst {
  focus: () => void
  blur: () => void
}

export type DatePickerProps = ExtractPublicPropTypes<typeof datePickerProps>

export type DatePickerClearSlotOnClear = () => void
export interface DatePickerClearSlotProps {
  onClear: DatePickerClearSlotOnClear
  text: string
}

export type DatePickerNowSlotOnNow = () => void
export interface DatePickerNowSlotProps {
  onNow: DatePickerNowSlotOnNow
  text: string
}

export type DatePickerConfirmSlotOnConfirm = () => void
export interface DatePickerConfirmSlotProps {
  onConfirm: DatePickerConfirmSlotOnConfirm
  disabled: boolean
  text: string
}
