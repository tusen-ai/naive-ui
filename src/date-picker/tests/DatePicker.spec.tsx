import { ref } from 'vue'
import { mount } from '@vue/test-utils'
import { format } from 'date-fns/esm'
import { NDatePicker } from '../index'
import { Value } from '../src/interface'
import { dateEnUS } from '../../locales'

describe('n-date-picker', () => {
  it('should work with import on demand', () => {
    mount(NDatePicker).unmount()
  })

  it('date type should work with shortcuts prop', async () => {
    const test = ref<[number, number]>([0, 0])
    const wrapper = mount(NDatePicker, {
      props: {
        value: test.value,
        type: 'date',
        onUpdateValue: (value: [number, number]) => {
          test.value = value
        },
        shortcuts: {
          'Honey birthday': 1631203200000
        }
      }
    })
    await wrapper.find('.n-input').trigger('click')
    const button: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    button.click()
    expect(test.value).toEqual(1631203200000)
    test.value = [0, 0]
    await wrapper.setProps({
      type: 'datetime'
    })
    await wrapper.find('.n-input').trigger('click')
    const timeButton: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    timeButton.click()
    expect(test.value).toEqual(1631203200000)
    wrapper.unmount()
  })

  it('range type should work with shortcuts prop', async () => {
    const test = ref<Value>(0)
    const wrapper = mount(NDatePicker, {
      props: {
        value: test.value,
        type: 'daterange',
        onUpdateValue: (value: Value) => {
          test.value = value
        },
        shortcuts: {
          'Honey birthday': [1629216000000, 1631203200000]
        }
      }
    })
    await wrapper.find('.n-input').trigger('click')
    const button: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    button.click()
    expect(test.value).toEqual([1629216000000, 1631203200000])
    test.value = 0
    await wrapper.setProps({
      type: 'datetimerange'
    })
    await wrapper.find('.n-input').trigger('click')
    const rangeButton: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    rangeButton.click()
    expect(test.value).toEqual([1629216000000, 1631203200000])
    wrapper.unmount()
  })

  it('date type should work with shortcuts prop with function value', async () => {
    const test = ref<Value>(0)
    const wrapper = mount(NDatePicker, {
      props: {
        value: test.value,
        type: 'date',
        onUpdateValue: (value: Value) => {
          test.value = value
        },
        shortcuts: {
          'Honey birthday': () => 1631203200000
        }
      }
    })
    await wrapper.find('.n-input').trigger('click')
    const button: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    button.click()
    expect(test.value).toEqual(1631203200000)
    test.value = 0
    await wrapper.setProps({
      type: 'datetime'
    })
    await wrapper.find('.n-input').trigger('click')
    const timeButton: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    timeButton.click()
    expect(test.value).toEqual(1631203200000)
    wrapper.unmount()
  })

  it('range type should work with shortcuts prop with function value', async () => {
    const test = ref<[number, number]>([0, 0])

    const wrapper = mount(NDatePicker, {
      props: {
        value: test.value,
        type: 'daterange',
        onUpdateValue: (value: [number, number]) => {
          test.value = value
        },
        shortcuts: {
          'Honey birthday': () =>
            [1629216000000, 1631203200000] as [number, number]
        }
      }
    })
    await wrapper.find('.n-input').trigger('click')
    const button: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    button.click()
    expect(test.value).toEqual([1629216000000, 1631203200000])
    test.value = [0, 0]
    await wrapper.setProps({
      type: 'datetimerange'
    })
    await wrapper.find('.n-input').trigger('click')
    const rangeButton: HTMLElement = document
      .querySelector('.n-date-panel-actions')
      ?.querySelector('.n-button') as HTMLElement
    rangeButton.click()
    expect(test.value).toEqual([1629216000000, 1631203200000])
    wrapper.unmount()
  })

  it('should work with `inputReadonly` prop', async () => {
    const wrapper = mount(NDatePicker)
    expect(wrapper.find('input').attributes('readonly')).not.toBe('')
    await wrapper.setProps({
      inputReadonly: true
    })
    expect(wrapper.find('input').attributes('readonly')).toBe('')
  })

  it('should work with `clearable` prop', async () => {
    const wrapper = mount(NDatePicker)
    expect(wrapper.find('.n-base-clear').exists()).not.toBe(true)
    await wrapper.setProps({
      clearable: true
    })
    expect(wrapper.find('.n-base-clear').exists()).toBe(true)
    wrapper.unmount()
  })

  it('should work with `disabled` prop', async () => {
    const wrapper = mount(NDatePicker)
    expect(wrapper.find('.n-input').attributes('class')).not.toContain(
      'n-input--disabled'
    )
    await wrapper.setProps({
      disabled: true
    })
    expect(wrapper.find('.n-input').attributes('class')).toContain(
      'n-input--disabled'
    )
    wrapper.unmount()
  })

  it('should work with `size` prop', async () => {
    ;(['small', 'medium', 'large'] as const).forEach((item) => {
      const wrapper = mount(NDatePicker, { props: { size: item } })
      expect(wrapper.find('.n-input').attributes('style')).toMatchSnapshot()
      wrapper.unmount()
    })
  })

  it('should work with `placeholder` prop', async () => {
    const wrapper = mount(NDatePicker)
    expect(wrapper.find('input').attributes('placeholder')).toBe('Select Date')
    await wrapper.setProps({
      placeholder: 'test-placeholder'
    })
    expect(wrapper.find('input').attributes('placeholder')).toBe(
      'test-placeholder'
    )
    wrapper.unmount()
  })

  it('should work with `defaultValue` prop', async () => {
    const wrapper = mount(NDatePicker, {
      props: {
        defaultValue: 1183135260000
      }
    })

    const inputEl = await wrapper.find('.n-input__input').find('input')
    expect(inputEl.element.value).toEqual(
      format(1183135260000, 'yyyy-MM-dd', {
        locale: dateEnUS.locale
      })
    )
  })

  it('should work with `firstDayOfWeek` prop', async () => {
    const wrapper = mount(NDatePicker, {
      attachTo: document.body,
      props: {
        firstDayOfWeek: 1
      }
    })

    await wrapper.find('.n-input__input').trigger('click')
    expect(
      document.querySelectorAll('.n-date-panel-weekdays__day')[0].textContent
    ).toBe('Tu')

    wrapper.unmount()
  })

  it('should work with `type` prop', async () => {
    const wrapper = mount(NDatePicker, {
      attachTo: document.body,
      props: { type: 'date' }
    })

    await wrapper.find('.n-input__input').trigger('click')
    expect(document.querySelector('.n-date-panel--date')).not.toEqual(null)

    await wrapper.setProps({ type: 'datetime' })
    await wrapper.find('.n-input__input').trigger('click')
    expect(document.querySelector('.n-date-panel--datetime')).not.toEqual(null)

    await wrapper.setProps({ type: 'daterange' })
    await wrapper.find('.n-input__input').trigger('click')
    expect(document.querySelector('.n-date-panel--daterange')).not.toEqual(null)

    await wrapper.setProps({ type: 'datetimerange' })
    await wrapper.find('.n-input__input').trigger('click')
    expect(document.querySelector('.n-date-panel--datetimerange')).not.toEqual(
      null
    )

    await wrapper.setProps({ type: 'month' })
    await wrapper.find('.n-input__input').trigger('click')
    expect(document.querySelector('.n-date-panel--month')).not.toEqual(null)

    wrapper.unmount()
  })

  it('should work with `onBlur` prop', async () => {
    const onBlur = vi.fn()
    const wrapper = mount(NDatePicker, {
      props: { onBlur }
    })

    await wrapper.find('input').trigger('focus')
    await wrapper.find('input').trigger('blur')
    expect(onBlur).toHaveBeenCalled()

    wrapper.unmount()
  })

  it('should work with `onFocus` prop', async () => {
    const onFocus = vi.fn()
    const wrapper = mount(NDatePicker, {
      props: { onFocus }
    })

    await wrapper.find('input').trigger('focus')
    expect(onFocus).toHaveBeenCalled()

    wrapper.unmount()
  })

  it('should work with `separator` prop', async () => {
    const wrapper = mount(NDatePicker, {
      props: { separator: '07akioni', type: 'daterange' }
    })
    expect(wrapper.text().includes('07akioni')).toBe(true)
    await wrapper.setProps({ separator: '08akioni', type: 'datetimerange' })
    expect(wrapper.text().includes('08akioni')).toBe(true)
  })

  it('should work with `status` prop', async () => {
    ;(['success', 'warning', 'error'] as const).forEach((status) => {
      const wrapper = mount(NDatePicker, {
        props: { status: status }
      })
      expect(wrapper.find('.n-input').classes()).toContain(
        `n-input--${status}-status`
      )
      wrapper.unmount()
    })
  })
})
