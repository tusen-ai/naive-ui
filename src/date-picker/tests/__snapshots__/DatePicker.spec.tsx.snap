// Vitest Snapshot v1

exports[`n-date-picker > should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-count-text-color: rgb(118, 124, 130); --n-color: rgba(255, 255, 255, 1); --n-font-size: 14px; --n-border-radius: 3px; --n-height: 28px; --n-padding-left: 10px; --n-padding-right: 10px; --n-text-color: rgb(51, 54, 57); --n-caret-color: #18a058; --n-text-decoration-color: rgb(51, 54, 57); --n-border: 1px solid rgb(224, 224, 230); --n-border-disabled: 1px solid rgb(224, 224, 230); --n-border-hover: 1px solid #36ad6a; --n-border-focus: 1px solid #36ad6a; --n-placeholder-color: rgba(194, 194, 194, 1); --n-placeholder-color-disabled: rgba(209, 209, 209, 1); --n-icon-size: 16px; --n-line-height-textarea: 1.6; --n-color-disabled: rgb(250, 250, 252); --n-color-focus: rgba(255, 255, 255, 1); --n-text-color-disabled: rgba(194, 194, 194, 1); --n-box-shadow-focus: 0 0 0 2px rgba(24, 160, 88, 0.2); --n-loading-color: #18a058; --n-caret-color-warning: #f0a020; --n-color-focus-warning: rgba(255, 255, 255, 1); --n-box-shadow-focus-warning: 0 0 0 2px rgba(240, 160, 32, 0.2); --n-border-warning: 1px solid #f0a020; --n-border-focus-warning: 1px solid #fcb040; --n-border-hover-warning: 1px solid #fcb040; --n-loading-color-warning: #f0a020; --n-caret-color-error: #d03050; --n-color-focus-error: rgba(255, 255, 255, 1); --n-box-shadow-focus-error: 0 0 0 2px rgba(208, 48, 80, 0.2); --n-border-error: 1px solid #d03050; --n-border-focus-error: 1px solid #de576d; --n-border-hover-error: 1px solid #de576d; --n-loading-color-error: #d03050; --n-clear-color: rgba(194, 194, 194, 1); --n-clear-size: 16px; --n-clear-color-hover: rgba(146, 146, 146, 1); --n-clear-color-pressed: rgba(175, 175, 175, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-hover: rgba(146, 146, 146, 1); --n-icon-color-pressed: rgba(175, 175, 175, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1); --n-suffix-text-color: rgb(51, 54, 57);"`;

exports[`n-date-picker > should work with \`size\` prop 2`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-count-text-color: rgb(118, 124, 130); --n-color: rgba(255, 255, 255, 1); --n-font-size: 14px; --n-border-radius: 3px; --n-height: 34px; --n-padding-left: 12px; --n-padding-right: 12px; --n-text-color: rgb(51, 54, 57); --n-caret-color: #18a058; --n-text-decoration-color: rgb(51, 54, 57); --n-border: 1px solid rgb(224, 224, 230); --n-border-disabled: 1px solid rgb(224, 224, 230); --n-border-hover: 1px solid #36ad6a; --n-border-focus: 1px solid #36ad6a; --n-placeholder-color: rgba(194, 194, 194, 1); --n-placeholder-color-disabled: rgba(209, 209, 209, 1); --n-icon-size: 16px; --n-line-height-textarea: 1.6; --n-color-disabled: rgb(250, 250, 252); --n-color-focus: rgba(255, 255, 255, 1); --n-text-color-disabled: rgba(194, 194, 194, 1); --n-box-shadow-focus: 0 0 0 2px rgba(24, 160, 88, 0.2); --n-loading-color: #18a058; --n-caret-color-warning: #f0a020; --n-color-focus-warning: rgba(255, 255, 255, 1); --n-box-shadow-focus-warning: 0 0 0 2px rgba(240, 160, 32, 0.2); --n-border-warning: 1px solid #f0a020; --n-border-focus-warning: 1px solid #fcb040; --n-border-hover-warning: 1px solid #fcb040; --n-loading-color-warning: #f0a020; --n-caret-color-error: #d03050; --n-color-focus-error: rgba(255, 255, 255, 1); --n-box-shadow-focus-error: 0 0 0 2px rgba(208, 48, 80, 0.2); --n-border-error: 1px solid #d03050; --n-border-focus-error: 1px solid #de576d; --n-border-hover-error: 1px solid #de576d; --n-loading-color-error: #d03050; --n-clear-color: rgba(194, 194, 194, 1); --n-clear-size: 16px; --n-clear-color-hover: rgba(146, 146, 146, 1); --n-clear-color-pressed: rgba(175, 175, 175, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-hover: rgba(146, 146, 146, 1); --n-icon-color-pressed: rgba(175, 175, 175, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1); --n-suffix-text-color: rgb(51, 54, 57);"`;

exports[`n-date-picker > should work with \`size\` prop 3`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-count-text-color: rgb(118, 124, 130); --n-color: rgba(255, 255, 255, 1); --n-font-size: 15px; --n-border-radius: 3px; --n-height: 40px; --n-padding-left: 14px; --n-padding-right: 14px; --n-text-color: rgb(51, 54, 57); --n-caret-color: #18a058; --n-text-decoration-color: rgb(51, 54, 57); --n-border: 1px solid rgb(224, 224, 230); --n-border-disabled: 1px solid rgb(224, 224, 230); --n-border-hover: 1px solid #36ad6a; --n-border-focus: 1px solid #36ad6a; --n-placeholder-color: rgba(194, 194, 194, 1); --n-placeholder-color-disabled: rgba(209, 209, 209, 1); --n-icon-size: 16px; --n-line-height-textarea: 1.6; --n-color-disabled: rgb(250, 250, 252); --n-color-focus: rgba(255, 255, 255, 1); --n-text-color-disabled: rgba(194, 194, 194, 1); --n-box-shadow-focus: 0 0 0 2px rgba(24, 160, 88, 0.2); --n-loading-color: #18a058; --n-caret-color-warning: #f0a020; --n-color-focus-warning: rgba(255, 255, 255, 1); --n-box-shadow-focus-warning: 0 0 0 2px rgba(240, 160, 32, 0.2); --n-border-warning: 1px solid #f0a020; --n-border-focus-warning: 1px solid #fcb040; --n-border-hover-warning: 1px solid #fcb040; --n-loading-color-warning: #f0a020; --n-caret-color-error: #d03050; --n-color-focus-error: rgba(255, 255, 255, 1); --n-box-shadow-focus-error: 0 0 0 2px rgba(208, 48, 80, 0.2); --n-border-error: 1px solid #d03050; --n-border-focus-error: 1px solid #de576d; --n-border-hover-error: 1px solid #de576d; --n-loading-color-error: #d03050; --n-clear-color: rgba(194, 194, 194, 1); --n-clear-size: 16px; --n-clear-color-hover: rgba(146, 146, 146, 1); --n-clear-color-pressed: rgba(175, 175, 175, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-hover: rgba(146, 146, 146, 1); --n-icon-color-pressed: rgba(175, 175, 175, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1); --n-suffix-text-color: rgb(51, 54, 57);"`;
